storybookjs/storybook
# CSF Tools: Add direct export features to export lists

Closes #22427



## What I did

- I extracted the parsing logic from `VariableDeclarator`, `FunctionDeclaration`, and `ExpressionStatement` nodes to reuse on nodes related to `ExportSpecifier` nodes.
- I extended the handling of `ExportSpecifier` nodes to gather information from nodes with the extracted parsing logic. The parent node is searched to find a `VariableDeclarator` or `FunctionDeclaration` with the same name of the `ExportSpecifier` node, then that node and any related `ExpressionStatement` nodes between until the `ExportSpecifier` node are also parsed.
- The `ExpressionStatement` nodes still check if the `name` property is default first and applies the logic already found there. If a related `VariableDeclarator` or `FunctionDeclaration` node is not found, the previous code that existed before is used as a fallback.

## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Open Storybook in your browser
3. Update stories to be exported via an export list instead of directly
4. No changes in behavior should be noticed

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)~

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
